scene/gui/rich_text_label.cpp

```diff
@@ -2990,8 +2998,13 @@ void RichTextLabel::add_image(const Ref<Texture2D> &p_image, const int p_width,
 			// calculate width to keep aspect ratio
 			item->size.width = p_image->get_width() * p_height / p_image->get_height();
 		} else {
-			// keep original width and height
-			item->size = p_image->get_size();
+			if (p_region.has_area()) {
```

Comment:
It probably should also use the `p_region.get_size()` in the two branches (L2992 and L2999) above as a base size for the aspect ratio calculation when region is set.

Reply:
done